pydy/bodies.py

```diff
@@ -0,0 +1,168 @@
+
+        if mass is None:
+            self._mass = Symbol(self.name + '_mass')
+        else:
+            self._mass = mass
+
+        if frame is None:
+            self._frame = ReferenceFrame(self.name + '_frame')
+        else:
+            self._frame = frame
+
+        if inertia is None:
+            inertia = outer(self._frame.x, self._frame.x)  # tensor product
+            self._inertia = (inertia, self._masscenter)
+        else:
+            self._inertia = inertia
+
+        # If user passes masscenter and mass then a particle is created
+        # otherwise a rigidbody. As a result a body may or may not have inertia.
+        if inertia is None and mass is not None:
+            self.body = Particle(self.name, self._masscenter, self._mass)
+        else:
+            self.body = RigidBody(self.name, self._masscenter, self._frame,
+                                  self._mass, self._inertia)
+
+    def add_coordinate(self, coordinate):
+        self._coordinates.append(coordinate)
+
+    def add_speed(self, speed):
+        self._speeds.append(speed)
+
+    def add_force(self, point_vector, force_vector):
+        """
+        Adds force to the body by adding Force's instance to the force_list.
+        force_list is used by system to get the tuples from the instance and
+        create force_list for Kane's method.
+
+        Parameters
+        ---------
+        point_vector: A 3 Tuple
+            Defines an arbitary point on a body on which the force will be
+            applied. The tuple defines the x, y and z values of the point
+            w.r.t body's masscenter.
+        force_vector: A 3 Tuple
+            Defines the force vector w.r.t frame of the body. The tuple defines
+            the values of vector in x, y and z directions of body's frame.
+
+        Example
+        --------
+        To add a force of magnitude 1 in y direction on a point at distance of
+        1 in x direction. All the directions are w.r.t body's frame.
+
+        >>> body = Body('body')
+        >>> body.add_force((1,0,0), (0,1,0))
+
+        """
+        point_vector = self.convert_tuple_into_vector(point_vector)
+        force_vector = self.convert_tuple_into_vector(force_vector)
+        point = self._masscenter.locatenew(self.name + '_point' + self._counter, point_vector)
+        self.force_list.append((point, force_vector))
+        self._counter += 1
+
+    def convert_tuple_into_vector(self, values_tuple):
+        vector = 0
+        vector += vector + values_tuple[0] * self.frame.x
+        vector += vector + values_tuple[1] * self.frame.y
+        vector += vector + values_tuple[2] * self.frame.z
+        return vector
+
+
+class Ground(Body):
+    """
+    A Ground Body
+
+    It has infinite mass and an origin.
+    """
+    def __init__(self):
+        name = 'ground'
+        masscenter = Point('origin')
+        mass = oo
+        # TODO Discuss if Ground needs to be RigidBody or Particle is fine?
```

Comment:
I'm not sure that we even need a special ground body in the first place. If you do need it, this infinite mass will not come into any computations, so I don't think it is needed at all.